File: HeartForCharity/HeartForCharity.Services/Database/VolunteerSkill.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace HeartForCharity.Services.Database
{

    public class VolunteerSkill
    {
        [Key]
        public int VolunteerSkillId { get; set; }

        [Required]
        [ForeignKey(nameof(UserProfile))]
        public int UserProfileId { get; set; }


        [Required]
        [ForeignKey(nameof(Skill))]
        public int SkillId { get; set; }

        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public virtual UserProfile UserProfile { get; set; } = null!;
        public virtual Skill Skill { get; set; } = null!;
    }
}

File: HeartForCharity/UI/heartforcharity_mobile/lib/model/responses/volunteer_skill.dart
class VolunteerSkill {
  final int volunteerSkillId;
  final int userProfileId;
  final int skillId;
  final String skillName;
  final String? skillDescription;

  VolunteerSkill({
    required this.volunteerSkillId,
    required this.userProfileId,
    required this.skillId,
    required this.skillName,
    this.skillDescription,
  });

  factory VolunteerSkill.fromJson(Map<String, dynamic> json) => VolunteerSkill(
        volunteerSkillId: json['volunteerSkillId'] ?? 0,
        userProfileId: json['userProfileId'] ?? 0,
        skillId: json['skillId'] ?? 0,
        skillName: json['skillName'] ?? '',
        skillDescription: json['skillDescription'],
      );
}
